src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -116,18 +118,29 @@ public Launcher decorate(final Launcher launcher, final Node node) {
             public Proc launch(ProcStarter starter) throws IOException {
                 boolean quiet = starter.quiet();
                 FilePath pwd = starter.pwd();
-                String[] commands = getCommands(starter);
 
-                return doLaunch(quiet, pwd, commands);
+                String [] cmdEnvs = starter.envs();
+
+                //check if the cmd is sourced from Jenkins, rather than another plugin; if so, skip cmdEnvs as we are getting other environment variables
+                for (String cmd : cmdEnvs) {
+                    if (cmd.contains(JENKINS_HOME)) {
```

Comment:
Mmh, can we be more specific here?

Another environment variable containing the string "JENKINS_HOME" might exist..

Reply:
The root of the issue is related not being able to determine the class that requested the proct  - when running a simple sh, you get the masters variables injected with no real way to tell you to ignore. If it's a plugin like the ssh-agent, you only see what cmdenvs that were passed along by the that class, which wouldn't include JENKINS_HOME.